swarm.go

```diff
@@ -182,7 +182,21 @@ func (s *Swarm) Process() goprocess.Process {
 }
 
 func (s *Swarm) addConn(tc transport.CapableConn, dir network.Direction) (*Conn, error) {
-	p := tc.RemotePeer()
+	var (
+		p    = tc.RemotePeer()
+		addr = tc.RemoteMultiaddr()
+	)
+
+	if s.gater != nil {
+		if allow := s.gater.InterceptAddrDial(p, addr); !allow {
```

Comment:
@raulk Why do we need this here ?

Reply:
https://github.com/libp2p/go-libp2p-swarm/pull/201#discussion_r425792282